src/comb_filter.rs

```diff
@@ -16,31 +22,89 @@ pub enum FilterParam {
 
 #[derive(Debug, Clone)]
 pub enum Error {
-    InvalidValue { param: FilterParam, value: f32 }
+    InvalidValue { param: FilterParam, value: f32 },
 }
 
 impl CombFilter {
     pub fn new(filter_type: FilterType, max_delay_secs: f32, sample_rate_hz: f32, num_channels: usize) -> Self {
-        todo!("implement")
+        let delay_samples = (sample_rate_hz * max_delay_secs) as usize;
+        let delay_buffers = vec![vec![0.0; delay_samples]; num_channels];
+        
+        CombFilter {
+            filter_type,
+            sample_rate_hz,
+            num_channels,
+            gain: 0.0, 
+            delay_secs: max_delay_secs,
+            delay_samples,
+            delay_buffers,
+        }
     }
 
     pub fn reset(&mut self) {
-        todo!("implement")
+        for buffer in &mut self.delay_buffers {
+            buffer.fill(0.0);
+        }
     }
 
     pub fn process(&mut self, input: &[&[f32]], output: &mut [&mut [f32]]) {
-        todo!("implement");
+        for (channel_idx, channel_buffers) in input.iter().enumerate() {
+            let delay_buffer = &mut self.delay_buffers[channel_idx];
+            let mut read_idx = self.delay_samples % delay_buffer.len(); 
```

Comment:
Because this variable is local to this loop inside the process function, it will never advance more than (block_size*channels) number of samples. I think this is why there's a big difference between your python and rust implementations. You will only every read and write from those samples. So when the delay is less than the block size times the number of channels this is fine, but with larger delays, this won't work. I'd suggest moving the read index to a field of your struct.